File: airport_challenge_javascript/src/Airport.js
'use strict';

function Airport() {
  this._hangar = [];
  this._MAX_CAPACITY = 10;
  this._CHANCE_OF_STORMY = 0.5;
};
Airport.prototype.planes = function(){
  return this._hangar;
};
Airport.prototype.clearForLanding = function(plane) {
  if(this._isFull()) {
    throw new Error('Cannot land plane. The airport is full.');
  }
  this._hangar.push(plane);
};
Airport.prototype.clearForTakeOff = function(plane) {
  if(this.isStormy()) {
    throw new Error('Cannot takeoff plane. The weather is stormy.');
  }
  this._hangar.pop();
};
Airport.prototype._isFull = function() {
  return this._hangar.length >= this._MAX_CAPACITY;
};
Airport.prototype.isStormy = function() {
  return (Math.random() > this._CHANCE_OF_STORMY);
};

File: week2-airport/airport_challenge-1/lib/airport.rb
class Airport
    def initialize(capacity)
      @capacity = capacity
      @airplanes = []
    end    
  
    def land(airplane)
      raise 'Airport full: Cannot land' if full? 
      raise 'Stormy: Cannot land' if stormy? 
      @airplanes << airplane 
    end
  
    def take_off(airplane)
      raise 'Cannot take off: weather is stormy' if stormy? 
    end
  end
  
  private 
  
  def full?
    @airplanes.length >= @capacity 
  end  
  
  def stormy?
    rand(1..10) > 8
  end   